### ISSUE TITLE
Parquet files

### ISSUE THREAD

**Issue by mskyttner** [r-lib/vroom#117]
Would reading parquet files be outside the scope of `vroom`? It is rectangular data.

https://stackoverflow.com/questions/50279914/how-to-read-a-parquet-file-in-r-without-using-spark-packages

**Comment by jimhester** [comment]
Yeah, binary formats are out of scope, vroom is for reading text formats only.

However the [arrow](https://github.com/apache/arrow/tree/master/r) R package has a `read_parquet()` function and I believe it has been recently submitted to CRAN.

**Comment by mskyttner** [comment]
Thanks for clarifying the scope of vroom. 

Regarding reading parquet files using `arrow`, support using a released version of the package may not be there yet... At least my attempts failed when trying it today on June 3 2019. 

My first attempt using the docker image "rocker/verse:3.60" to install `arrow` like so:

```
> install.packages("arrow")
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
Warning in install.packages :
  package ‘arrow’ is not available (for R version 3.6.0)
```

I also tried installing the latest release version of the arrow R package from https://github.com/apache/arrow/tree/master/r which provides install instructions and is based on the Apache Arrow release 0.13 from April which involved building it using this Dockerfile:

```
FROM rocker/verse:3.6.0

# add support for Apache Arrow and reading parquet files using "arrow" R package

RUN rm -vfr /var/lib/apt/lists/*

RUN apt-get update --fix-missing && apt-get install -y -V \
	apt-transport-https curl gnupg lsb-release

RUN printf "deb http://deb.debian.org/debian $(lsb_release --codename --short)-backports main" > /etc/apt/sources.list.d/backports.list

RUN curl --output /usr/share/keyrings/apache-arrow-keyring.gpg https://dl.bintray.com/apache/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-keyring.gpg

RUN printf "deb [arch=amd64 signed-by=/usr/share/keyrings/apache-arrow-keyring.gpg] https://dl.bintray.com/apache/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/ $(lsb_release --codename --short) main \ndeb-src [signed-by=/usr/share/keyrings/apache-arrow-keyring.gpg] https://dl.bintray.com/apache/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/ $(lsb_release --codename --short) main" > /etc/apt/sources.list.d/apache-arrow.list

RUN curl https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -

RUN printf "deb http://apt.llvm.org/$(lsb_release --codename --short)/ llvm-toolchain-$(lsb_release --codename --short)-7 main\ndeb-src http://apt.llvm.org/$(lsb_release --codename --short)/ llvm-toolchain-$(lsb_release --codename --short)-7 main" > /etc/apt/sources.list.d/llvm.list

RUN apt-get update && apt-get install -y -V \
	software-properties-common && \
	apt-add-repository contrib && \
	apt-add-repository non-free && \
	apt-get update && \
	apt-get install --no-install-recommends -y -V \
	libarrow-dev \
	libarrow-glib-dev \
	libplasma-dev \
	libplasma-glib-dev \
	libgandiva-dev \
	libgandiva-glib-dev \
	libparquet-dev \
	libparquet-glib-dev

RUN R -e 'remotes::install_github("apache/arrow/r", ref="76e1bc5dfb9d08e31eddd5cbcc0b1bab934da2c7")' 

RUN rm -vfr /var/lib/apt/lists/* && rm -rf /tmp/downloaded_packages/ /tmp/*.rds && \
	apt-get autoclean && \
	apt-get autoremove -y
```

Using this docker image, my second attempt also failed with:

```r
library(arrow)
df <- read_parquet("artdata.parquet")
Error in read_parquet_file(file) : Support for Parquet is not available.

# df <- read_parquet(system.file("v0.7.1.parquet", package="arrow"))
# this also cannot read the bundled parquet file with the same error message
```

I finally tried to install the `arrow` package with `devtools::install_github("apache/arrow/r")` as [this SO post](https://stackoverflow.com/questions/30402253/how-do-i-read-a-parquet-in-r-and-convert-it-to-an-r-dataframe) indicated that it should work, however for me that too failed, this time with a compilation error (`/usr/local/lib/R/etc/Makeconf:176: recipe for target 'array_from_vector.o' failed`). 

So it seems this functionality is in the works but not available in any released version yet as of today?

I can see that [it has been done with Apache Arrow before](https://gist.github.com/yutannihilation/5ec19ea26775d20a7242db3eb28872b9) so I guess some patience is needed until it gets there. 

It would be great to have a nice interface similar to what you offer for reading files using `vroom` - I guess this is inspired from `readr` - preferably something that allows for reading individual columns only perhaps even with "pushdown" so that data can be collected into memory only when needed.... 

Do you think it makes sense to have  similar interface for reading parquet files? If so I could make a wish to have this in https://github.com/apache/arrow/blob/master/r/R/parquet.R over at https://github.com/apache/arrow/issues. 

Meanwhile it looks like the `sparklyr::spark_read_parquet` works but will it will require an Apache Spark installation.

**Comment by xhochy** [comment]
> Would reading parquet files be outside the scope of `vroom`? It is rectangular data.

Small clarification here: Parquet is for tabular data, i.e. everything happens on a per-column base. For data frames, it's a perfect fit but for multidimensional matrices, I recommend to look out for alternative.
